core/server/apps/amp/lib/router.js

```diff
@@ -42,21 +43,42 @@ function getPostData(req, res, next) {
             next();
         })
         .catch(function (err) {
-            if (err instanceof errors.NotFoundError) {
-                return next(err);
-            }
-
             next(err);
         });
 }
 
+function checkIfAMPIsEnabled(req, res, next) {
+    var ampIsEnabled = config.theme.amp;
+
+    if (ampIsEnabled) {
+        return next();
+    }
+
+    // CASE: we don't support amp pages for static pages
+    if (req.body.post && req.body.post.page) {
+        return errors.error404(req, res, next);
+    }
+
+    /**
+     * CASE: amp is disabled, we redirect the user to the requested post
```

Comment:
https://support.google.com/webmasters/answer/181708?hl=en

Essentially if a page or requested representation of a page doesn't exist then the response code should always be 404. Sending any other success (2/3xx) code tells search engines that the page or representation does exist which can result in SEO problems, phantom pages in indexes, and errors because the resulting page is not in the requested format (in this case AMP is requested but we return HTML)